### DISCUSSION TITLE
Logo proposition for the organization

### DISCUSSION THREAD

**Discussion by Its-Just-Nans** [node-saml/node-saml#166] [Ideas]
Here is a proposition of a logo for the organization `node-saml` https://github.com/node-saml

![b](https://user-images.githubusercontent.com/56606507/188307112-4e6ed5ad-9528-4076-914a-f07df37d706c.png)

**Comment by srd90** [comment]
_Note/edit: This comment was written when logo proposition contained nodejs logo with additional ”saml” text. Author of the proposition has since then changed/edited proposed logo image. This comment along with links to trademark stuff is not deleted so that others may evaluate new / upcoming propositions against some trademark policy documents._

---

First of all: I am not a lawyer.

Just out of curiosity: Is that logo a-ok from possible registered trademarks point of view?

Found following documents which might be related to js logo used in your proposition:

1. https://nodejs.org/static/documents/trademark-policy.pdf
2. https://openjsf.org/blog/2022/02/14/node-js-trademarks-transferred-to-openjs-foundation/
3. https://openjsf.org/wp-content/uploads/sites/84/2021/01/OpenJS-Foundation-Trademark-Policy-2021-01-12.docx.pdf
    * https://nodejs.org/en/  's ”Trademark policy” link points to link 3's document 4 September 2022

Link 1's document might or might not be overrided by link 3's document due to trademark transfer (see link 2 and make your own interpretation).

Again: I am not a lawyer. I just listed some of the links that I found (4 September 2022) with https://www.google.com/search?q=nodejs%20logo%20trademark

**Comment by cjbarth** [comment]
@Its-Just-Nans , I also remember a different logo. I certainly don't want to run afoul of any trademark or copyright issues, but it would be nice if you'd include some reference that this is version 2 or something so that we can keep track. Changing the first post with no indication that you've done so is a little misleading and confusing, an potentially creates additional work for very helpful contributors like @srd90 .

**Comment by Its-Just-Nans** [comment]
Yes

Here is the two versions:

V1
![a](https://user-images.githubusercontent.com/56606507/188321962-214a78c7-2a80-47c1-9167-83d8ff1b9001.png)

V2
![b](https://user-images.githubusercontent.com/56606507/188321969-4813d2d2-0a4a-4452-9988-6de9d22774a8.png)

**Reply by markstos** [reply]
I think this logo is OK. The green hexagon evokes the JavaScript logo, while SAML in the Handel Gothic font may be something some people recognize from other places where SAML has been presented that way. The color green is associated with safety, which is appropriate. 

These shades of green could look OK on a white or black background, which will be happening as people use light mode and dark mode color schemes. (I'm looking at it on Github's dark mode). 

Personally, I don't like the Handel Gothic font and might prefer a blue color to evoke security and trust, but for the reasons above, the current treatment is a sensible choice. Because of the nature of SAML, I think creating an exciting and appropriate logo for it is a hard task. 

Here's one idea for a refinement to consider:

  1. Swap the greens for blues
  2. Move the SAML lettering down so it aligns with the bottom of the vertical sides of the hex
  3. Add a simple padlock icon above SAML to communicate it's a authentication technology. That would help people understand a little more about what SAML is at a glance. 


More on color symbolism is here: https://www.color-meanings.com/color-symbolism-chart/

**Reply by cjbarth** [reply]
@Its-Just-Nans , would you be willing to make the above adjustments and post another sample?

**Reply by Its-Just-Nans** [reply]
Something like this ?

![logo](https://user-images.githubusercontent.com/56606507/202524551-66f8406e-36ac-41a2-ae3f-4bd9e9fea003.png)

**Reply by markstos** [reply]
This looks good to me!  It looks better on a white background than Github's dark background, but it's hard to look great in both.

In HTML, it's possible to specify alternate light and dark mode logos, but I prefer the simplicity of using a single logo.  

I could vote for this version.

**Reply by cjbarth** [reply]
@Its-Just-Nans , what copyright do you have over this (Creative Commons maybe)? Or are you thinking of turning over the copyright to the @node-saml organization?

**Reply by Its-Just-Nans** [reply]
As you want, both are good for me

**Reply by cjbarth** [reply]
@markstos , I don't have a lot of experiencing with licensing. I'm inclined toward a Creative Commons license, but if this is to be just for `node-saml`, then maybe that isn't appropriate since others could then use it. Do you have experience in this area?

**Reply by markstos** [reply]
@cjbarth You are right to flag this  case as special because it's a logo and not source code. The way that Mozilla Firefox dealt with this was to apply an open source license to the logo files themselves, but apply a trademark to the images and words as branding. This how some Linux distributions ended up with "Ice Fox" or "Ice Weasel" as browser names. The issue was that the Firefox trademark could only be applied to the official distribution as it was distributed by Mozilla. Because Debian modified the software, the trademark prevented them from calling the modified version Firefox or using the same logo, which could have led to confusion that it was the same product or that Mozilla endorsed the changes. It seems like we have the same kind of situation here. It seems fine if people were to fork the source files for this, but it could be bad for security if another project, especially a malicious project, were allowed to use the same logo and claim it was "node-saml" or "passport-saml".  The Mozilla trademark policy page is here: https://www.mozilla.org/en-US/foundation/trademarks/policy/ But honestly, I would welcome some guidance scaling down what Mozilla did to a tiny project of our size. It seems like there should be a way to handle this that's not complicated.

**Reply by Its-Just-Nans** [reply]
I have no experience in that, both are good for me ;)

**Comment by cjbarth** [comment]
@markstos , what do you think of this?

**Reply by markstos** [reply]
Besides the resolved concerns about the JS logo, writing "SAML" in this way is used in other places. It doesn't appear with the registered trademark symbol, but the origin is unclear. For example, you can see SAML in this font here:

https://www.pingidentity.com/en/platform/capabilities/open-standard-protocols.html

I'd feel more comfortable if I knew the origin of writing SAML in this font.

**Reply by cjbarth** [reply]
The comment about the font is a valid observation. Based on my scan of that "SAML" graphic using [WhatTheFont](https://www.myfonts.com/pages/whatthefont), it seems that the logo is just the letters S A M L typed out in [Handel Gothic](https://en.wikipedia.org/wiki/Handel_Gothic). You can try it out [here](https://www.myfonts.com/collections/handel-gothic-font-bitstream) and see.

Additionally, based on [this](http://saml.xml.org/wiki/saml-logos) and [this](https://dribbble.com/shots/4041061-SAML-Logo) post, there is no official logo for SAML right now. I certainly couldn't find one anywhere. What is more, googling for "saml logo" shows images of all sorts, whereas googling for "openid logo" shows a very consistent logo.

**Reply by srd90** [reply]
This document seems to be talking something about usage of SAML text (I do not understand english and/or law stuff well enough to understand if/when/where/how it can be used):
https://www.oasis-open.org/policies-guidelines/trademark/
(referenced 14 November 2022)

**Reply by cjbarth** [reply]
From that document @srd90 , it seems we can just put the following at the bottom of our README and we're good:

> “[OASIS](http://www.oasis-open.org/)”, [“SAML” and “Security Assertion Markup Language” ] are trademarks of OASIS, the open standards consortium where the [SAML] specification is owned and developed. [SAML] is a copyrighted © work of OASIS Open. All rights reserved.
